Upgrade intermediate menus from main page:

GiudGiud opened the pull request:
- add more links
- sort things into categories for app development
- tell people to use AD
refs #22234

GiudGiud committed:
Upgrade intermediate menus from main page:
- add more links
- sort things into categories
- tell people to use AD
refs #22234

GiudGiud commented on modules/doc/content/application_development/jacobian_definition.md:
```diff
@@ -7,3 +7,7 @@
+
+!alert note
+We recommend using Automatic Differentiation for most new developments. Hand-coding the Jacobian
+should only be considered for performance reasons.
```
@lindsayad fyi

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1210616/) on 8c8dbdd wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22457/site)

This comment will be updated on new commits.

GiudGiud committed:
Add explicit link to
- physics modules page
- python tools pages
as downstream apps that are including the index of app_development will never include all the modules docs and that would leave broken links

hugary1995 commented:
(While you are at this,) I'll say this upfront: the wire twisting animation on the main page is absolutely fantastic -- I know how difficult that simulation is. But... I feel like it is not the best strategy to show the single-physics wire twisting animation, while we are advertising MOOSE as a **multi**physics framework.

GiudGiud commented:
Should pass now
